### ISSUE TITLE
Experment result in dgcnn-cls

### ISSUE THREAD

**Issue by tsbiosky** [yichen928/STRL#6]
Thanks for your great work! I tried to run the training experiment with dgcnn in classification task .But I only got around 70% accuarcy in SVM evaluation and 89.6% in fine-tune evaluation. During training I found my val_loss is lower than 0.2 after 10 epoch , but the checkpoint you provided has 0.513 val_loss at epoch 51 . Is there something wrong with my config ? Thanks

optimizer:
    weight_decay: 0.01
    lr: 1e-3
    type: adam

network: DGCNN
dataset: ShapeNet    #   ShapeNet, ShapeNetPart, ModelNet40, ScanNet
num_points: 2048    #   2048 for ShapeNet, 4096 for ModelNet40, 4096 for ScanNet
epochs: 100
batch_size: 32
acc_batches: 1
transform_mode: both

decay_rate: 0.996
mlp_hidden_size: 4096
projection_size: 256

k: 40
emb_dims: 1024
window_length: 3
dropout: 0.5
num_workers: 32

resume_ckpt:

**Comment by yichen928** [comment]
Hi, thank you for your interest!

The weight decay should be zero. I am sorry for this mistake in the config file. I have fixed it in the repo.

In fact, there is no notable correlation between the val_loss and model performance, and the val_loss may not continue decreasing as the training goes.  In most cases, the performance would be better if you train the model longer.

**Comment by tsbiosky** [comment]
> 

Thank you for your update .
But I tried the updated dgcnn-cls config and I still got a very low acc(<70%) in SVM .
I also tried Pointnet-cls with default config but I only get around 30% acc in SVM .
I'm very confused because I didn't change any code .

DGCNN config:
{'optimizer.weight_decay': 0.0, 'optimizer.lr': 0.001, 'optimizer.type': 'adam', 'task_model.class': 'BYOL.models.BasicalModel', 'task_model.name': 'BYOL-cls-dgcnn', 'distrib_backend': 'dp', 'network': 'DGCNN', 'dataset': 'ShapeNet', 'num_points': 2048, 'epochs': 100, 'batch_size': 32, 'acc_batches': 1, 'transform_mode': 'both', 'decay_rate': 0.996, 'mlp_hidden_size': 4096, 'projection_size': 256, 'k': 40, 'emb_dims': 1024, 'window_length': 3, 'dropout': 0.5, 'num_workers': 32}

Pointnet config:
{'optimizer.weight_decay': 0.0, 'optimizer.lr': 0.001, 'optimizer.type': 'adam', 'task_model.class': 'BYOL.models.BasicalModel', 'task_model.name': 'BYOL-cls-PointNet', 'distrib_backend': 'dp', 'network': 'PointNet', 'dataset': 'ShapeNet', 'num_points': 2048, 'epochs': 50, 'batch_size': 48, 'acc_batches': 1, 'transform_mode': 'spatial', 'decay_rate': 0.996, 'mlp_hidden_size': 4096, 'projection_size': 256, 'emb_dims': 1024, 'num_workers': 32}

**Comment by yichen928** [comment]
Hi, I ran the training code for PointNet, and got a linear classifier result similar to that in the paper after about only 10 epochs. This is my config:

{'optimizer.weight_decay': 0.0, 'optimizer.lr': 0.001, 'optimizer.type': 'adam', 'task_model.class': 'BYOL.models.BasicalModel', 'task_model.name': 'BYOL-cls-PointNet', 'distrib_backend': 'dp', 'network': 'PointNet', 'dataset': 'ShapeNet', 'num_points': 2048, 'epochs': 50, 'batch_size': 48, 'acc_batches': 1, 'transform_mode': 'both', 'decay_rate': 0.996, 'mlp_hidden_size': 4096, 'projection_size': 256, 'emb_dims': 1024, 'num_workers': 32}

You may notice that the transform_mode should be "both", but actually this only marginally affects the performance. I am not sure what happened to you but you may double-check the data.
